# História das Olimpíadas

Após um ano de atraso por conta da pandemia de Covid-19, as atenções do mundo todo se voltaram para Tóquio, no Japão, para acompanhar mais uma edição das Olimpíadas.

No Brasil não foi diferente, e muitos se uniram para torcer por nossos atletas em diferentes competições, tanto em esportes onde o Brasil já possui tradição quanto em novos esportes.

Vamos aproveitar o clima para estudar um pouco das Olimpíadas! Utilizaremos um _dataset_ com 120 anos de dados históricos das Olimpíadas, cobrindo desde os jogos de Atenas 1896 até Rio 2016. 

Faça o download do _dataset_ em https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results e carregue o arquivo ```athlete_events.csv``` para um DataFrame utilizando Pandas. Aproveite para explorar seu DataFrame e se familiarizar com a sua estrutura. 

OBS: Fique à vontade para acrescentar mais células Python conforme necessário em qualquer etapa do exercício.

In [1]:
import pandas as pd
import numpy as np
pd.options.display.float_format = '{:,.2f}'.format

df_olimp = pd.read_csv('athlete_events.csv')

df_olimp

ID                      Name Sex   Age  Height  Weight  \
0            1                 A Dijiang   M 24.00  180.00   80.00   
1            2                  A Lamusi   M 23.00  170.00   60.00   
2            3       Gunnar Nielsen Aaby   M 24.00     NaN     NaN   
3            4      Edgar Lindenau Aabye   M 34.00     NaN     NaN   
4            5  Christine Jacoba Aaftink   F 21.00  185.00   82.00   
...        ...                       ...  ..   ...     ...     ...   
271111  135569                Andrzej ya   M 29.00  179.00   89.00   
271112  135570                  Piotr ya   M 27.00  176.00   59.00   
271113  135570                  Piotr ya   M 27.00  176.00   59.00   
271114  135571        Tomasz Ireneusz ya   M 30.00  185.00   96.00   
271115  135571        Tomasz Ireneusz ya   M 34.00  185.00   96.00   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  
0          Basketball               Basketball Men's Basketball   NaN  
1                Judo              Judo Men's Extra-Lightweight   NaN  
2            Football                   Football Men's Football   NaN  
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold  
4       Speed Skating          Speed Skating Women's 500 metres   NaN  
...               ...                                       ...   ...  
271111           Luge                Luge Mixed (Men)'s Doubles   NaN  
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual   NaN  
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team   NaN  
271114      Bobsleigh                      Bobsleigh Men's Four   NaN  
271115      Bobsleigh                      Bobsleigh Men's Four   NaN  

[271116 rows x 15 columns]

## 1. O Brasil nas Olimpíadas

Vamos começar estudando o desempenho do nossos próprio país. Gere um DataFrame novo contendo apenas as informações sobre atletas brasileiros.

In [2]:
df_brazil = df_olimp[df_olimp['NOC'] == 'BRA']
df_brazil

ID                        Name Sex   Age  Height  Weight    Team  \
665        386             Alexandre Abeid   M 22.00  194.00   92.00  Brazil   
666        386             Alexandre Abeid   M 26.00  194.00   92.00  Brazil   
668        388  Abel Carlos da Silva Braga   M 19.00  190.00   73.00  Brazil   
781        451         Diana Monteiro Abla   F 21.00  175.00   75.00  Brazil   
1005       565        Glauclio Serro Abreu   M 26.00  185.00   75.00  Brazil   
...        ...                         ...  ..   ...     ...     ...     ...   
268652  134400   Maria Jos "Zz" Bertolotti   F 26.00  190.00     NaN  Brazil   
269745  134976       Sieglinda Lenk Zigler   F 17.00     NaN     NaN  Brazil   
269899  135037           Moacir Zimmermann   M 32.00  184.00   70.00  Brazil   
270478  135289        Zzimo Alves Calazans   M 20.00     NaN     NaN  Brazil   
270789  135423        Ral Benedito Zumbano   M 22.00     NaN     NaN  Brazil   

        NOC        Games  Year  Season            City       Sport  \
665     BRA  1972 Summer  1972  Summer          Munich  Volleyball   
666     BRA  1976 Summer  1976  Summer        Montreal  Volleyball   
668     BRA  1972 Summer  1972  Summer          Munich    Football   
781     BRA  2016 Summer  2016  Summer  Rio de Janeiro  Water Polo   
1005    BRA  2004 Summer  2004  Summer          Athina      Boxing   
...     ...          ...   ...     ...             ...         ...   
268652  BRA  1992 Summer  1992  Summer       Barcelona  Basketball   
269745  BRA  1936 Summer  1936  Summer          Berlin    Swimming   
269899  BRA  2016 Summer  2016  Summer  Rio de Janeiro   Athletics   
270478  BRA  1952 Summer  1952  Summer        Helsinki    Football   
270789  BRA  1948 Summer  1948  Summer          London      Boxing   

                                         Event Medal  
665                Volleyball Men's Volleyball   NaN  
666                Volleyball Men's Volleyball   NaN  
668                    Football Men's Football   NaN  
781              Water Polo Women's Water Polo   NaN  
1005                 Boxing Men's Middleweight   NaN  
...                                        ...   ...  
268652           Basketball Women's Basketball   NaN  
269745  Swimming Women's 100 metres Backstroke   NaN  
269899      Athletics Men's 20 kilometres Walk   NaN  
270478                 Football Men's Football   NaN  
270789                Boxing Men's Lightweight   NaN  

[3848 rows x 15 columns]

### Medalhistas

Vamos focar um pouco nos casos de sucesso do Brasil. Use o seu DataFrame anterior para filtrar apenas informações sobre **medalhistas** brasileiros. 

**DICA:** observe como a coluna ```Medal``` é representada quando o atleta não ganhou medalha.

In [3]:
df_brazil_medal = df_brazil[~df_brazil['Medal'].isna()]

df_brazil_medal

ID                                          Name Sex   Age  \
1651       918                           Ademir Roque Kaefer   M 24.00   
1652       918                           Ademir Roque Kaefer   M 28.00   
1668       925           Adenzia Aparecida Ferreira da Silva   F 25.00   
1733       966                                  Daniel Adler   M 26.00   
1856      1020                  Adriana Aparecida dos Santos   F 25.00   
...        ...                                           ...  ..   ...   
267709  133931                      Arthur Nabarrete Zanetti   M 26.00   
268211  134186              Jos Elias "Z Elias" Moedim Jnior   M 19.00   
268218  134190  Jos Marco "Z Marco" Nobrega Ferreira de Melo   M 29.00   
268219  134191                Jos Marcelo "Z Maria" Ferreira   M 22.00   
268247  134211                        Jos Carlos Cracco Neto   M 22.00   

        Height  Weight      Team  NOC        Games  Year  Season  \
1651    179.00   74.00    Brazil  BRA  1984 Summer  1984  Summer   
1652    179.00   74.00    Brazil  BRA  1988 Summer  1988  Summer   
1668    187.00   65.00    Brazil  BRA  2012 Summer  2012  Summer   
1733    180.00   72.00    Brazil  BRA  1984 Summer  1984  Summer   
1856    180.00   61.00    Brazil  BRA  1996 Summer  1996  Summer   
...        ...     ...       ...  ...          ...   ...     ...   
267709  156.00   61.00    Brazil  BRA  2016 Summer  2016  Summer   
268211  178.00   75.00    Brazil  BRA  1996 Summer  1996  Summer   
268218  190.00   86.00  Brazil-1  BRA  2000 Summer  2000  Summer   
268219  175.00   71.00    Brazil  BRA  1996 Summer  1996  Summer   
268247  170.00   69.00    Brazil  BRA  2016 Summer  2016  Summer   

                  City             Sport  \
1651       Los Angeles          Football   
1652             Seoul          Football   
1668            London        Volleyball   
1733       Los Angeles           Sailing   
1856           Atlanta        Basketball   
...                ...               ...   
267709  Rio de Janeiro        Gymnastics   
268211         Atlanta          Football   
268218          Sydney  Beach Volleyball   
268219         Atlanta          Football   
268247  Rio de Janeiro          Football   

                                          Event   Medal  
1651                    Football Men's Football  Silver  
1652                    Football Men's Football  Silver  
1668              Volleyball Women's Volleyball    Gold  
1733        Sailing Mixed Three Person Keelboat  Silver  
1856              Basketball Women's Basketball  Silver  
...                                         ...     ...  
267709                   Gymnastics Men's Rings  Silver  
268211                  Football Men's Football  Bronze  
268218  Beach Volleyball Men's Beach Volleyball  Silver  
268219                  Football Men's Football  Bronze  
268247                  Football Men's Football    Gold  

[475 rows x 15 columns]

### Verão vs Inverno

Você deve ter notado que temos duas categorias distintas de jogos olímpicos, representados pela estação: temos os jogos de verão e os jogos de inverno, que ocorrem de maneira intercalada.

Agora que já conhecemos os medalhistas brasileiros, resposta: quantos atletas brasileiros receberam medalha nos jogos de verão e quantos receberam nos jogos de inverno?

In [4]:
df_bmv = df_brazil_medal[df_brazil_medal['Season'] == 'Summer']
print('Esses foram os medalhistas brasileiros dos jogos de verão:',len(df_bmv))


df_bmi = df_brazil_medal[df_brazil_medal['Season'] == 'Winter']
print('Esses foram os medalhistas brasileiros dos jogos de inverno:',len(df_bmi))

Esses foram os medalhistas brasileiros dos jogos de verão: 475
Esses foram os medalhistas brasileiros dos jogos de inverno: 0


Os jogos de verão são bem mais populares do que os jogos de inverno no Brasil. Portanto, deste ponto em diante iremos focar apenas nos jogos de verão. Descarte de seu DataFrame os dados dos jogos de inverno.



In [5]:
df_brazil_medal = df_bmv

df_brazil_medal

ID                                          Name Sex   Age  \
1651       918                           Ademir Roque Kaefer   M 24.00   
1652       918                           Ademir Roque Kaefer   M 28.00   
1668       925           Adenzia Aparecida Ferreira da Silva   F 25.00   
1733       966                                  Daniel Adler   M 26.00   
1856      1020                  Adriana Aparecida dos Santos   F 25.00   
...        ...                                           ...  ..   ...   
267709  133931                      Arthur Nabarrete Zanetti   M 26.00   
268211  134186              Jos Elias "Z Elias" Moedim Jnior   M 19.00   
268218  134190  Jos Marco "Z Marco" Nobrega Ferreira de Melo   M 29.00   
268219  134191                Jos Marcelo "Z Maria" Ferreira   M 22.00   
268247  134211                        Jos Carlos Cracco Neto   M 22.00   

        Height  Weight      Team  NOC        Games  Year  Season  \
1651    179.00   74.00    Brazil  BRA  1984 Summer  1984  Summer   
1652    179.00   74.00    Brazil  BRA  1988 Summer  1988  Summer   
1668    187.00   65.00    Brazil  BRA  2012 Summer  2012  Summer   
1733    180.00   72.00    Brazil  BRA  1984 Summer  1984  Summer   
1856    180.00   61.00    Brazil  BRA  1996 Summer  1996  Summer   
...        ...     ...       ...  ...          ...   ...     ...   
267709  156.00   61.00    Brazil  BRA  2016 Summer  2016  Summer   
268211  178.00   75.00    Brazil  BRA  1996 Summer  1996  Summer   
268218  190.00   86.00  Brazil-1  BRA  2000 Summer  2000  Summer   
268219  175.00   71.00    Brazil  BRA  1996 Summer  1996  Summer   
268247  170.00   69.00    Brazil  BRA  2016 Summer  2016  Summer   

                  City             Sport  \
1651       Los Angeles          Football   
1652             Seoul          Football   
1668            London        Volleyball   
1733       Los Angeles           Sailing   
1856           Atlanta        Basketball   
...                ...               ...   
267709  Rio de Janeiro        Gymnastics   
268211         Atlanta          Football   
268218          Sydney  Beach Volleyball   
268219         Atlanta          Football   
268247  Rio de Janeiro          Football   

                                          Event   Medal  
1651                    Football Men's Football  Silver  
1652                    Football Men's Football  Silver  
1668              Volleyball Women's Volleyball    Gold  
1733        Sailing Mixed Three Person Keelboat  Silver  
1856              Basketball Women's Basketball  Silver  
...                                         ...     ...  
267709                   Gymnastics Men's Rings  Silver  
268211                  Football Men's Football  Bronze  
268218  Beach Volleyball Men's Beach Volleyball  Silver  
268219                  Football Men's Football  Bronze  
268247                  Football Men's Football    Gold  

[475 rows x 15 columns]

### Atletas do Brasil

Vamos conhecer um pouco melhor nossos atletas. Descubra a altura e peso médio de nossos medalhistas.

In [6]:
print('Essa é a altura média dos medalhistas brasileiros:',df_brazil_medal['Height'].mean())

print('Essa é o peso médio dos medalhistas brasileiros:',df_brazil_medal['Weight'].mean())

Essa é a altura média dos medalhistas brasileiros: 182.4945295404814
Essa é o peso médio dos medalhistas brasileiros: 76.7069351230425


Imaginamos que diferentes esportes podem beneficiar diferentes tipos físicos, certo? Então refaça a análise anterior, mas obtendo os valores médios **por esporte**.

In [7]:
print('Essa é a altura média dos medalhistas brasileiros, por esporte:',df_brazil_medal.groupby(['Sport'])['Height'].mean())

print('Essa é o peso médio dos medalhistas brasileiros, por esporte:',df_brazil_medal.groupby(['Sport'])['Weight'].mean())

Essa é a altura média dos medalhistas brasileiros, por esporte: Sport
Athletics           181.00
Basketball          185.61
Beach Volleyball    184.88
Boxing              170.00
Canoeing            175.00
Equestrianism       179.67
Football            175.80
Gymnastics          162.75
Judo                176.67
Modern Pentathlon   166.00
Sailing             181.59
Shooting            175.00
Swimming            189.11
Taekwondo           184.00
Volleyball          190.59
Name: Height, dtype: float64
Essa é o peso médio dos medalhistas brasileiros, por esporte: Sport
Athletics           74.58
Basketball          78.48
Beach Volleyball    78.16
Boxing              64.00
Canoeing            83.25
Equestrianism       75.00
Football            69.96
Gymnastics          63.75
Judo                86.29
Modern Pentathlon   55.00
Sailing             80.41
Shooting            69.00
Swimming            81.56
Taekwondo           79.50
Volleyball          81.17
Name: Weight, dtype: float64


Será que os dados acima influenciaram no interesse geral dos atletas pelo esporte ou realmente impactaram no desempenho deles? Podemos tentar descobrir se há algum tipo de correlação.

Você ainda possui o dataframe original contendo todos os atletas brasileiros, incluindo os sem medalha? Obtenha os valores médios de peso e altura por esporte daquele dataframe e compare-o com os dos medalhistas. Há alguma diferença significativa em algum esporte?

In [8]:
print('Essa é a altura média dos brasileiros, por esporte:',df_brazil.groupby(['Sport'])['Height'].mean())

print('Essa é o peso médio dos brasileiros, por esporte:',df_brazil.groupby(['Sport'])['Weight'].mean())

Essa é a altura média dos brasileiros, por esporte: Sport
Alpine Skiing           176.63
Archery                 172.60
Art Competitions           NaN
Athletics               176.20
Badminton               175.50
Basketball              190.91
Beach Volleyball        185.70
Biathlon                170.00
Bobsleigh               182.64
Boxing                  171.99
Canoeing                177.79
Cross Country Skiing    173.00
Cycling                 174.24
Diving                  167.44
Equestrianism           177.43
Fencing                 179.20
Figure Skating          156.00
Football                173.37
Freestyle Skiing        154.00
Golf                    170.00
Gymnastics              157.27
Handball                181.47
Hockey                  177.44
Judo                    173.93
Luge                    175.00
Modern Pentathlon       168.60
Rhythmic Gymnastics     164.16
Rowing                  180.55
Rugby Sevens            171.24
Sailing                 179.68
Shooting    

In [9]:
print('Essa é a diferença de altura média dos medalhistas em relação a altura média de todos os atletas brasileiros:\n',(df_brazil_medal.groupby(['Sport'])['Height'].mean()) - (df_brazil.groupby(['Sport'])['Height'].mean()))

Essa é a diferença de altura média dos medalhistas em relação a altura média de todos os atletas brasileiros:
 Sport
Alpine Skiing             NaN
Archery                   NaN
Art Competitions          NaN
Athletics                4.80
Badminton                 NaN
Basketball              -5.30
Beach Volleyball        -0.82
Biathlon                  NaN
Bobsleigh                 NaN
Boxing                  -1.99
Canoeing                -2.79
Cross Country Skiing      NaN
Cycling                   NaN
Diving                    NaN
Equestrianism            2.24
Fencing                   NaN
Figure Skating            NaN
Football                 2.43
Freestyle Skiing          NaN
Golf                      NaN
Gymnastics               5.48
Handball                  NaN
Hockey                    NaN
Judo                     2.74
Luge                      NaN
Modern Pentathlon       -2.60
Rhythmic Gymnastics       NaN
Rowing                    NaN
Rugby Sevens              NaN
Sailing      

Vemos alguns esportes onde as alturas médias são menores do que a média geral. As maiores diferenças desse grupo são:
- Basquete (-5.30);
- Canoagem (-2.79);
- Pentathlon Moderno (-2.60).

E temos grandes disparidades em alguns esportes, onde a altura média dos medalhistas é bem representativa, como:
- Atletismo (4.80);
- Ginástica olímpica (5.48);
- Natação (7.93);
- Taekwondo (10.23).

In [10]:
print('Essa é a diferença de peso médio dos medalhistas em relação ao peso médio de todos os atletas brasileiros:\n',(df_brazil_medal.groupby(['Sport'])['Weight'].mean()) - (df_brazil.groupby(['Sport'])['Weight'].mean()))

Essa é a diferença de peso médio dos medalhistas em relação ao peso médio de todos os atletas brasileiros:
 Sport
Alpine Skiing             NaN
Archery                   NaN
Art Competitions          NaN
Athletics                6.78
Badminton                 NaN
Basketball              -7.42
Beach Volleyball        -0.46
Biathlon                  NaN
Bobsleigh                 NaN
Boxing                  -0.11
Canoeing                 5.85
Cross Country Skiing      NaN
Cycling                   NaN
Diving                    NaN
Equestrianism            2.69
Fencing                   NaN
Figure Skating            NaN
Football                 2.20
Freestyle Skiing          NaN
Golf                      NaN
Gymnastics              11.29
Handball                  NaN
Hockey                    NaN
Judo                     6.00
Luge                      NaN
Modern Pentathlon       -5.67
Rhythmic Gymnastics       NaN
Rowing                    NaN
Rugby Sevens              NaN
Sailing         

Quanto ao peso médio, temos como as maiores diferenças onde a média dos pesos dos medalhistas é menor nos esportes são:
- Basquete (-7.42);
- Pentathlon Moderno(-5.67);
- Tiro ao Alvo (-7.92).

Agora, as maiores diferenças onde a média dos pesos dos medalhistas é maior nos esportes são:
- Atletismo (6.78);
- Canoagem (5.85);
- Judô (6.00);
- Natação (6.97);
- Taekwondo (15.88).

Existe um detalhe importante passando batido até agora em nossa análise: as categorias esportivas costumam ser divididas por gênero justamente por conta de diferenças físicas entre homens e mulheres que poderiam influenciar no desempenho. Compare a altura e peso médios de atletas brasileiros por esporte segmentado por sexo.

In [11]:
print('Essa é a diferença de altura média dos medalhistas em relação a altura média de todos os atletas brasileiros, por sexo:\n',(df_brazil_medal.groupby(['Sex', 'Sport'])['Height'].mean()) - (df_brazil.groupby(['Sport'])['Height'].mean()))

print('Essa é a diferença de peso médio dos medalhistas em relação ao peso médio de todos os atletas brasileiros, por sexo:\n',(df_brazil_medal.groupby(['Sex', 'Sport'])['Weight'].mean()) - (df_brazil.groupby(['Sport'])['Weight'].mean()))

Essa é a diferença de altura média dos medalhistas em relação a altura média de todos os atletas brasileiros, por sexo:
 Sex  Sport            
F    Athletics            -3.20
     Basketball           -7.54
     Beach Volleyball    -10.70
     Boxing               -4.99
     Football             -6.40
     Judo                 -5.73
     Modern Pentathlon    -2.60
     Sailing              -9.18
     Swimming            -16.18
     Taekwondo             4.23
     Volleyball           -4.32
M    Athletics             5.12
     Basketball           -3.32
     Beach Volleyball      9.88
     Boxing               -1.24
     Canoeing             -2.79
     Equestrianism         2.24
     Football              5.62
     Gymnastics            5.48
     Judo                  5.39
     Sailing               3.39
     Shooting              0.96
     Swimming              9.27
     Taekwondo            16.23
     Volleyball            7.60
Name: Height, dtype: float64
Essa é a diferença de peso 

Qual foi (ou quais foram) o maior medalhista brasileiro em quantidade total de medalhas?

In [12]:
df_bm = df_brazil_medal.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games', 'Year', 'Season', 'City', 'Sport', 'Event'], axis=1)

In [13]:
df_bm

ID                                          Name   Medal
1651       918                           Ademir Roque Kaefer  Silver
1652       918                           Ademir Roque Kaefer  Silver
1668       925           Adenzia Aparecida Ferreira da Silva    Gold
1733       966                                  Daniel Adler  Silver
1856      1020                  Adriana Aparecida dos Santos  Silver
...        ...                                           ...     ...
267709  133931                      Arthur Nabarrete Zanetti  Silver
268211  134186              Jos Elias "Z Elias" Moedim Jnior  Bronze
268218  134190  Jos Marco "Z Marco" Nobrega Ferreira de Melo  Silver
268219  134191                Jos Marcelo "Z Maria" Ferreira  Bronze
268247  134211                        Jos Carlos Cracco Neto    Gold

[475 rows x 3 columns]

In [15]:
df_bm.groupby(['ID'])['Medal'].count().reset_index()

ID  Medal
0       918      2
1       925      1
2       966      1
3      1020      2
4      1023      1
..      ...    ...
355  133931      2
356  134186      1
357  134190      1
358  134191      1
359  134211      1

[360 rows x 2 columns]

In [16]:
df_bm.groupby(['ID']).sum()

Name         Medal
ID                                                                     
918                Ademir Roque KaeferAdemir Roque Kaefer  SilverSilver
925                   Adenzia Aparecida Ferreira da Silva          Gold
966                                          Daniel Adler        Silver
1020    Adriana Aparecida dos SantosAdriana Aparecida ...  SilverBronze
1023                     Adriana "Adrianinha" Moiss Pinto        Bronze
...                                                   ...           ...
133931   Arthur Nabarrete ZanettiArthur Nabarrete Zanetti    GoldSilver
134186                   Jos Elias "Z Elias" Moedim Jnior        Bronze
134190       Jos Marco "Z Marco" Nobrega Ferreira de Melo        Silver
134191                     Jos Marcelo "Z Maria" Ferreira        Bronze
134211                             Jos Carlos Cracco Neto          Gold

[360 rows x 2 columns]

In [17]:
ID_maismed = df_bm.groupby(['ID']).size()[df_bm.groupby(['ID']).size() == df_bm.groupby(['ID']).size().max()].reset_index()
ID_maismed

ID  0
0   42229  5
1  106683  5

In [18]:
print('Os maiores medalhistas brasileiros em quantidade total de medalhas foram:\n',df_bm[df_bm['ID'].isin(ID_maismed['ID'])]['Name'].drop_duplicates().reset_index()['Name'].tolist())

Os maiores medalhistas brasileiros em quantidade total de medalhas foram:
 ['Torben Schmidt Grael', 'Robert Scheidt']


In [19]:
print(df_bm[df_bm['ID'] == 42229].iloc[0,1])

print(df_bm[df_bm['ID'] == 106683].iloc[0,1])

Torben Schmidt Grael
Robert Scheidt


E o(s) maior(es) em quantidade de medalhas de ouro?

In [20]:
df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size()

ID
925       1
2758      1
3330      1
3790      1
5093      1
         ..
129573    1
129899    1
130541    1
133931    1
134211    1
Length: 96, dtype: int64

In [21]:
df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size().max()

2

In [22]:
df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size()[(df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size()).values == 2].count()

13

In [23]:
ID_ouro = df_bm[df_bm['Medal'] == 'Gold']. groupby(['ID']).size()[(df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size()).values == df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size().max()].keys().tolist()
ID_ouro

[24678,
 33095,
 33630,
 33638,
 34995,
 40494,
 42229,
 53867,
 76846,
 92177,
 106683,
 109482,
 119466]

In [24]:
print('Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:\n',df_bm[df_bm['ID'].isin(ID_ouro)]['Name'].drop_duplicates().reset_index()['Name'].tolist())

Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:
 ['Adhemar Ferreira da Silva', 'Srgio "Escadinha" Dutra dos Santos', 'Fabiana "Fabi" Alvim de Oliveira', 'Fabiana Marcelino Claudino', 'Marcelo Bastos Ferreira', 'Giovane Farinazzo Gvio', 'Torben Schmidt Grael', 'Jaqueline Maria "Jaque" Pereira de Carvalho Endres', 'Maurcio Camargo Lima', 'Paula Renata Marques Pequeno', 'Robert Scheidt', 'Sheilla Tavares de Castro Blassioli', 'Thasa Daher de Menezes']


In [25]:
print(df_bm[df_bm['ID'] == 24678].iloc[0,1])
print(df_bm[df_bm['ID'] == 33095].iloc[0,1])
print(df_bm[df_bm['ID'] == 33630].iloc[0,1])
print(df_bm[df_bm['ID'] == 33638].iloc[0,1])
print(df_bm[df_bm['ID'] == 34995].iloc[0,1])
print(df_bm[df_bm['ID'] == 40494].iloc[0,1])
print(df_bm[df_bm['ID'] == 42229].iloc[0,1])
print(df_bm[df_bm['ID'] == 53867].iloc[0,1])
print(df_bm[df_bm['ID'] == 76846].iloc[0,1])
print(df_bm[df_bm['ID'] == 92177].iloc[0,1])
print(df_bm[df_bm['ID'] == 106683].iloc[0,1])
print(df_bm[df_bm['ID'] == 109482].iloc[0,1])
print(df_bm[df_bm['ID'] == 119466].iloc[0,1])

Adhemar Ferreira da Silva
Srgio "Escadinha" Dutra dos Santos
Fabiana "Fabi" Alvim de Oliveira
Fabiana Marcelino Claudino
Marcelo Bastos Ferreira
Giovane Farinazzo Gvio
Torben Schmidt Grael
Jaqueline Maria "Jaque" Pereira de Carvalho Endres
Maurcio Camargo Lima
Paula Renata Marques Pequeno
Robert Scheidt
Sheilla Tavares de Castro Blassioli
Thasa Daher de Menezes


Qual esporte rendeu mais medalhas de ouro para o Brasil? E qual rendeu mais medalhas no total?

**DICA:** tome muito cuidado nessa análise: cada **evento esportivo** rende 1 medalha. Por exemplo, quando a equipe de futebol vence, isso é considerado 1 medalha, mesmo tendo cerca de 20 atletas medalhistas na equipe. 

In [26]:
df_sport = df_brazil_medal.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games', 'Season', 'City', 'Event', 'Name'], axis=1)

df_sport

ID  Year             Sport   Medal
1651       918  1984          Football  Silver
1652       918  1988          Football  Silver
1668       925  2012        Volleyball    Gold
1733       966  1984           Sailing  Silver
1856      1020  1996        Basketball  Silver
...        ...   ...               ...     ...
267709  133931  2016        Gymnastics  Silver
268211  134186  1996          Football  Bronze
268218  134190  2000  Beach Volleyball  Silver
268219  134191  1996          Football  Bronze
268247  134211  2016          Football    Gold

[475 rows x 4 columns]

In [27]:
df_sport.drop_duplicates(['Year', 'Sport'])

ID  Year             Sport   Medal
1651       918  1984          Football  Silver
1652       918  1988          Football  Silver
1668       925  2012        Volleyball    Gold
1733       966  1984           Sailing  Silver
1856      1020  1996        Basketball  Silver
...        ...   ...               ...     ...
194967   97845  2016          Canoeing  Bronze
202453  101690  1996  Beach Volleyball  Silver
221565  111358  2016         Taekwondo  Bronze
263339  131784  2016          Shooting  Silver
267707  133931  2012        Gymnastics    Gold

[84 rows x 4 columns]

In [28]:
df_sport.drop_duplicates(['Year', 'Sport'])[df_sport['Medal'] == 'Gold']

<ipython-input-28-6ac87f7bf71d>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_sport.drop_duplicates(['Year', 'Sport'])[df_sport['Medal'] == 'Gold']


ID  Year             Sport Medal
1668       925  2012        Volleyball  Gold
4825      2758  2016  Beach Volleyball  Gold
5930      3330  1992        Volleyball  Gold
6760      3790  2004        Volleyball  Gold
9260      5093  2016        Volleyball  Gold
22613    11920  1980           Sailing  Gold
28565    14810  2016         Athletics  Gold
35353    18204  1992              Judo  Gold
41794    21544  2008          Swimming  Gold
43889    22623  2016            Boxing  Gold
46586    24043  1984         Athletics  Gold
47882    24678  1956         Athletics  Gold
58064    29713  2016          Football  Gold
68551    34995  1996           Sailing  Gold
68553    34995  2004           Sailing  Gold
83124    42228  2016           Sailing  Gold
158468   79547  1988              Judo  Gold
186456   93746  2004     Equestrianism  Gold
267707  133931  2012        Gymnastics  Gold

In [29]:
df_sportouro = df_sport.drop_duplicates(['Year', 'Sport'])[df_sport['Medal'] == 'Gold'].groupby(['Sport'])['Medal'].count().reset_index()
df_sportouro

<ipython-input-29-e104211b1efe>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_sportouro = df_sport.drop_duplicates(['Year', 'Sport'])[df_sport['Medal'] == 'Gold'].groupby(['Sport'])['Medal'].count().reset_index()


Sport  Medal
0         Athletics      3
1  Beach Volleyball      1
2            Boxing      1
3     Equestrianism      1
4          Football      1
5        Gymnastics      1
6              Judo      2
7           Sailing      4
8          Swimming      1
9        Volleyball      4

In [30]:
print('Esses foram os esportes que mais renderam medalhas de ouro para o Brasil:\n',df_sportouro[df_sportouro['Medal'] == df_sportouro['Medal'].max()]['Sport'].tolist())

Esses foram os esportes que mais renderam medalhas de ouro para o Brasil:
 ['Sailing', 'Volleyball']


In [31]:
df_sportmed = df_sport.drop_duplicates(['Year', 'Sport']).groupby(['Sport'])['Medal'].count().reset_index()

df_sportmed

Sport  Medal
0           Athletics     13
1          Basketball      5
2    Beach Volleyball      6
3              Boxing      3
4            Canoeing      1
5       Equestrianism      3
6            Football      7
7          Gymnastics      2
8                Judo     10
9   Modern Pentathlon      1
10            Sailing     11
11           Shooting      2
12           Swimming     10
13          Taekwondo      2
14         Volleyball      8

In [32]:
print('Esse foi o esporte que mais rendeu medalhas em quantidade para o Brasil:\n',df_sportmed[df_sportmed['Medal'] == df_sportmed['Medal'].max()]['Sport'].tolist())

Esse foi o esporte que mais rendeu medalhas em quantidade para o Brasil:
 ['Athletics']


Cada "categoria" dentro de um esporte é considerado um evento. Por exemplo, dentro de "atletismo", temos uma competição de 100m masculina, uma de 100m feminino, um revezamento 4 x 100m masculino, um revezamento 4 x 100m feminino, uma competição de 400m masculino, uma de 400m feminino, uma maratona masculina, uma maratona feminina, e assim sucessivamente.

Sabendo disso, qual evento esportivo mais rendeu medalhas de ouro para o Brasil? E total de medalhas?

In [33]:
df_event = df_brazil_medal.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games', 'Season', 'City', 'Sport', 'Name'], axis=1)

df_event

ID  Year                                    Event   Medal
1651       918  1984                  Football Men's Football  Silver
1652       918  1988                  Football Men's Football  Silver
1668       925  2012            Volleyball Women's Volleyball    Gold
1733       966  1984      Sailing Mixed Three Person Keelboat  Silver
1856      1020  1996            Basketball Women's Basketball  Silver
...        ...   ...                                      ...     ...
267709  133931  2016                   Gymnastics Men's Rings  Silver
268211  134186  1996                  Football Men's Football  Bronze
268218  134190  2000  Beach Volleyball Men's Beach Volleyball  Silver
268219  134191  1996                  Football Men's Football  Bronze
268247  134211  2016                  Football Men's Football    Gold

[475 rows x 4 columns]

In [34]:
df_eventouro = df_event.drop_duplicates(['Year', 'Event'])[df_event['Medal'] == 'Gold'].groupby(['Event'])['Medal'].count().reset_index()
df_eventouro

<ipython-input-34-a8f8b207b9b3>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_eventouro = df_event.drop_duplicates(['Year', 'Event'])[df_event['Medal'] == 'Gold'].groupby(['Event'])['Medal'].count().reset_index()


Event  Medal
0                  Athletics Men's 800 metres      1
1                  Athletics Men's Pole Vault      1
2                 Athletics Men's Triple Jump      2
3                 Athletics Women's Long Jump      1
4     Beach Volleyball Men's Beach Volleyball      2
5                    Boxing Men's Lightweight      1
6     Equestrianism Mixed Jumping, Individual      1
7                     Football Men's Football      1
8                      Gymnastics Men's Rings      1
9                 Judo Men's Half-Heavyweight      1
10                Judo Men's Half-Lightweight      1
11             Judo Women's Extra-Lightweight      1
12                   Judo Women's Lightweight      1
13          Sailing Men's Two Person Keelboat      1
14                    Sailing Mixed Multihull      1
15            Sailing Mixed One Person Dinghy      2
16            Sailing Mixed Two Person Dinghy      1
17          Sailing Mixed Two Person Keelboat      1
18                      Sailing Women's Skiff      1
19  Shooting Men's Military Pistol, 30 metres      1
20         Swimming Men's 50 metres Freestyle      1
21                Volleyball Men's Volleyball      3
22              Volleyball Women's Volleyball      2

In [35]:
print('Esse foi o evento que mais rendeu medalhas de ouro para o Brasil:\n',df_eventouro[df_eventouro['Medal'] == df_eventouro['Medal'].max()]['Event'].tolist())

Esse foi o evento que mais rendeu medalhas de ouro para o Brasil:
 ["Volleyball Men's Volleyball"]


In [36]:
df_eventmed = df_event.drop_duplicates(['Year', 'Event']).groupby(['Event'])['Medal'].count().reset_index()

df_eventmed

Event  Medal
0                  Athletics Men's 200 metres      1
1        Athletics Men's 4 x 100 metres Relay      3
2                  Athletics Men's 800 metres      2
3                   Athletics Men's High Jump      1
4                    Athletics Men's Marathon      1
..                                        ...    ...
57  Swimming Women's 10 kilometres Open Water      1
58                Taekwondo Men's Heavyweight      1
59              Taekwondo Women's Heavyweight      1
60                Volleyball Men's Volleyball      6
61              Volleyball Women's Volleyball      4

[62 rows x 2 columns]

In [37]:
print('Esses foram os eventos que mais renderam medalhas em quantidade para o Brasil:\n',df_eventmed[df_eventmed['Medal'] == df_eventmed['Medal'].max()]['Event'].tolist())

Esses foram os eventos que mais renderam medalhas em quantidade para o Brasil:
 ["Athletics Men's Triple Jump", "Football Men's Football", "Volleyball Men's Volleyball"]


Para finalizar sobre o Brasil: obtenha o total de medalhas de ouro, prata, bronze e total por ano.

In [38]:
bm_medalano = df_sport.groupby(['Year', 'Medal'])['ID'].size().to_frame(name = 'Total').reset_index()

bm_medalano

Year   Medal  Total
0   1920  Bronze      5
1   1920    Gold      1
2   1920  Silver      1
3   1948  Bronze     10
4   1952  Bronze      2
5   1952    Gold      1
6   1956    Gold      1
7   1960  Bronze     13
8   1964  Bronze     12
9   1968  Bronze      3
10  1968  Silver      1
11  1972  Bronze      2
12  1976  Bronze      3
13  1980  Bronze      5
14  1980    Gold      4
15  1984  Bronze      2
16  1984    Gold      1
17  1984  Silver     33
18  1988  Bronze      5
19  1988    Gold      1
20  1988  Silver     17
21  1992    Gold     13
22  1992  Silver      1
23  1996  Bronze     43
24  1996    Gold      5
25  1996  Silver     15
26  2000  Bronze     36
27  2000  Silver     12
28  2004  Bronze      3
29  2004    Gold     18
30  2004  Silver     19
31  2008  Bronze     30
32  2008    Gold     14
33  2008  Silver     34
34  2012  Bronze     11
35  2012    Gold     14
36  2012  Silver     34
37  2016  Bronze      6
38  2016    Gold     36
39  2016  Silver      8

In [39]:
bm_medalano_pivot = bm_medalano.pivot(index = 'Year', columns = 'Medal', values = 'Total')

bm_medalano_pivot = bm_medalano_pivot.reindex(columns = ['Gold', 'Silver', 'Bronze'])

bm_medalano_pivot[['Gold', 'Silver', 'Bronze']] = bm_medalano_pivot[['Gold', 'Silver', 'Bronze']].fillna(value = 0)

bm_medalano_pivot

Medal  Gold  Silver  Bronze
Year                       
1920   1.00    1.00    5.00
1948   0.00    0.00   10.00
1952   1.00    0.00    2.00
1956   1.00    0.00    0.00
1960   0.00    0.00   13.00
1964   0.00    0.00   12.00
1968   0.00    1.00    3.00
1972   0.00    0.00    2.00
1976   0.00    0.00    3.00
1980   4.00    0.00    5.00
1984   1.00   33.00    2.00
1988   1.00   17.00    5.00
1992  13.00    1.00    0.00
1996   5.00   15.00   43.00
2000   0.00   12.00   36.00
2004  18.00   19.00    3.00
2008  14.00   34.00   30.00
2012  14.00   34.00   11.00
2016  36.00    8.00    6.00

## 2. O mundo nos jogos de verão

Vamos agora analisar um pouquinho do que aconteceu nas Olimpíadas de verão em todo o mundo.

Retome o DataFrame original e descarte as informações sobre os jogos de inverno.

In [40]:
df_verao = df_olimp[df_olimp['Season'] == 'Summer']

df_verao

ID                                Name Sex   Age  Height  Weight  \
0            1                           A Dijiang   M 24.00  180.00   80.00   
1            2                            A Lamusi   M 23.00  170.00   60.00   
2            3                 Gunnar Nielsen Aaby   M 24.00     NaN     NaN   
3            4                Edgar Lindenau Aabye   M 34.00     NaN     NaN   
26           8  Cornelia "Cor" Aalten (-Strannood)   F 18.00  168.00     NaN   
...        ...                                 ...  ..   ...     ...     ...   
271106  135565            Fernando scar Zylberberg   M 27.00  168.00   76.00   
271107  135566          James Francis "Jim" Zylker   M 21.00  175.00   75.00   
271108  135567        Aleksandr Viktorovich Zyuzin   M 24.00  183.00   72.00   
271109  135567        Aleksandr Viktorovich Zyuzin   M 28.00  183.00   72.00   
271110  135568              Olga Igorevna Zyuzkova   F 33.00  171.00   69.00   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
26         Netherlands  NED  1932 Summer  1932  Summer     Los Angeles   
...                ...  ...          ...   ...     ...             ...   
271106       Argentina  ARG  2004 Summer  2004  Summer          Athina   
271107   United States  USA  1972 Summer  1972  Summer          Munich   
271108          Russia  RUS  2000 Summer  2000  Summer          Sydney   
271109          Russia  RUS  2004 Summer  2004  Summer          Athina   
271110         Belarus  BLR  2016 Summer  2016  Summer  Rio de Janeiro   

             Sport                                   Event Medal  
0       Basketball             Basketball Men's Basketball   NaN  
1             Judo            Judo Men's Extra-Lightweight   NaN  
2         Football                 Football Men's Football   NaN  
3       Tug-Of-War             Tug-Of-War Men's Tug-Of-War  Gold  
26       Athletics            Athletics Women's 100 metres   NaN  
...            ...                                     ...   ...  
271106      Hockey                     Hockey Men's Hockey   NaN  
271107    Football                 Football Men's Football   NaN  
271108      Rowing  Rowing Men's Lightweight Coxless Fours   NaN  
271109      Rowing  Rowing Men's Lightweight Coxless Fours   NaN  
271110  Basketball           Basketball Women's Basketball   NaN  

[222552 rows x 15 columns]

Obtenha a lista de todos os esportes já disputados nas olimpíadas de verão.

In [41]:
print(df_verao.groupby(['Sport']).size().reset_index()['Sport'].tolist())

['Aeronautics', 'Alpinism', 'Archery', 'Art Competitions', 'Athletics', 'Badminton', 'Baseball', 'Basketball', 'Basque Pelota', 'Beach Volleyball', 'Boxing', 'Canoeing', 'Cricket', 'Croquet', 'Cycling', 'Diving', 'Equestrianism', 'Fencing', 'Figure Skating', 'Football', 'Golf', 'Gymnastics', 'Handball', 'Hockey', 'Ice Hockey', 'Jeu De Paume', 'Judo', 'Lacrosse', 'Modern Pentathlon', 'Motorboating', 'Polo', 'Racquets', 'Rhythmic Gymnastics', 'Roque', 'Rowing', 'Rugby', 'Rugby Sevens', 'Sailing', 'Shooting', 'Softball', 'Swimming', 'Synchronized Swimming', 'Table Tennis', 'Taekwondo', 'Tennis', 'Trampolining', 'Triathlon', 'Tug-Of-War', 'Volleyball', 'Water Polo', 'Weightlifting', 'Wrestling']


Obtenha a lista de todas as modalidades esportivas já disputadas nas olimpíadas de verão.

In [42]:
print(df_verao.groupby(['Event']).size().reset_index()['Event'].tolist())

['Aeronautics Mixed Aeronautics', 'Alpinism Mixed Alpinism', "Archery Men's Au Chapelet, 33 metres", "Archery Men's Au Chapelet, 50 metres", "Archery Men's Au Cordon Dore, 33 metres", "Archery Men's Au Cordon Dore, 50 metres", "Archery Men's Championnat Du Monde", "Archery Men's Continental Style", "Archery Men's Double American Round", "Archery Men's Double York Round", "Archery Men's Individual", "Archery Men's Pole Archery, Large Birds, Individual", "Archery Men's Pole Archery, Large Birds, Team", "Archery Men's Pole Archery, Small Birds, Individual", "Archery Men's Pole Archery, Small Birds, Team", "Archery Men's Sur La Perche a La Herse", "Archery Men's Sur La Perche a La Pyramide", "Archery Men's Target Archery, 28 metres, Individual", "Archery Men's Target Archery, 28 metres, Team", "Archery Men's Target Archery, 33 metres, Individual", "Archery Men's Target Archery, 33 metres, Team", "Archery Men's Target Archery, 50 metres, Individual", "Archery Men's Target Archery, 50 metres

Obtenha a lista de todos os países que já disputaram olimpíadas.

In [43]:
print(df_verao.groupby(['NOC']).size().reset_index()['NOC'].tolist())

['AFG', 'AHO', 'ALB', 'ALG', 'AND', 'ANG', 'ANT', 'ANZ', 'ARG', 'ARM', 'ARU', 'ASA', 'AUS', 'AUT', 'AZE', 'BAH', 'BAN', 'BAR', 'BDI', 'BEL', 'BEN', 'BER', 'BHU', 'BIH', 'BIZ', 'BLR', 'BOH', 'BOL', 'BOT', 'BRA', 'BRN', 'BRU', 'BUL', 'BUR', 'CAF', 'CAM', 'CAN', 'CAY', 'CGO', 'CHA', 'CHI', 'CHN', 'CIV', 'CMR', 'COD', 'COK', 'COL', 'COM', 'CPV', 'CRC', 'CRO', 'CRT', 'CUB', 'CYP', 'CZE', 'DEN', 'DJI', 'DMA', 'DOM', 'ECU', 'EGY', 'ERI', 'ESA', 'ESP', 'EST', 'ETH', 'EUN', 'FIJ', 'FIN', 'FRA', 'FRG', 'FSM', 'GAB', 'GAM', 'GBR', 'GBS', 'GDR', 'GEO', 'GEQ', 'GER', 'GHA', 'GRE', 'GRN', 'GUA', 'GUI', 'GUM', 'GUY', 'HAI', 'HKG', 'HON', 'HUN', 'INA', 'IND', 'IOA', 'IRI', 'IRL', 'IRQ', 'ISL', 'ISR', 'ISV', 'ITA', 'IVB', 'JAM', 'JOR', 'JPN', 'KAZ', 'KEN', 'KGZ', 'KIR', 'KOR', 'KOS', 'KSA', 'KUW', 'LAO', 'LAT', 'LBA', 'LBR', 'LCA', 'LES', 'LIB', 'LIE', 'LTU', 'LUX', 'MAD', 'MAL', 'MAR', 'MAS', 'MAW', 'MDA', 'MDV', 'MEX', 'MGL', 'MHL', 'MKD', 'MLI', 'MLT', 'MNE', 'MON', 'MOZ', 'MRI', 'MTN', 'MYA', 'NAM'

Qual atleta foi o maior medalhista (em medalhas totais) da história das olimpíadas de verão?

In [44]:
ID_medmundo = df_verao.groupby(['ID']).size()[df_verao.groupby(['ID']).size() == df_verao.groupby(['ID']).size().max()].reset_index()
ID_medmundo

ID   0
0  77710  58

In [45]:
print('O maiores medalhista do mundo em quantidade total de medalhas das olimpíadas de verão foi:\n',df_verao[df_verao['ID'].isin(ID_medmundo['ID'])]['Name'].drop_duplicates().reset_index()['Name'].tolist())

O maiores medalhista do mundo em quantidade total de medalhas das olimpíadas de verão foi:
 ['Robert Tait McKenzie']


Qual atleta foi o maior medalhista de ouro da história das olimpíadas de verão?

In [46]:
ID_ouromundo = df_verao[df_verao['Medal'] == 'Gold']. groupby(['ID']).size()[(df_verao[df_verao['Medal'] == 'Gold'].groupby(['ID']).size()).values == df_verao[df_verao['Medal'] == 'Gold'].groupby(['ID']).size().max()].keys().tolist()
ID_ouromundo

[94406]

In [47]:
print('O maior medalhista do mundo em quantidade total de medalhas de ouro nas olimpíadas de verão foi:\n',df_verao[df_verao['ID'].isin(ID_ouromundo)]['Name'].drop_duplicates().reset_index()['Name'].tolist())

O maior medalhista do mundo em quantidade total de medalhas de ouro nas olimpíadas de verão foi:
 ['Michael Fred Phelps, II']


Qual país foi o maior medalhista de ouro da história das olimpíadas de verão? Lembre-se da questão do evento esportivo, para não considerar múltiplas medalhas para um mesmo evento (ex: uma equipe de futebol fazendo parecer que mais de 20 medalhas foram distribuídas).

In [48]:
df_medal_mundo = df_verao[~df_verao['Medal'].isna()]

df_medal_mundo

ID                                 Name Sex   Age  Height  Weight  \
3            4                 Edgar Lindenau Aabye   M 34.00     NaN     NaN   
37          15                 Arvo Ossian Aaltonen   M 30.00     NaN     NaN   
38          15                 Arvo Ossian Aaltonen   M 30.00     NaN     NaN   
41          17              Paavo Johannes Aaltonen   M 28.00  175.00   64.00   
42          17              Paavo Johannes Aaltonen   M 28.00  175.00   64.00   
...        ...                                  ...  ..   ...     ...     ...   
271078  135553  Galina Ivanovna Zybina (-Fyodorova)   F 25.00  168.00   80.00   
271080  135553  Galina Ivanovna Zybina (-Fyodorova)   F 33.00  168.00   80.00   
271082  135554                         Bogusaw Zych   M 28.00  182.00   82.00   
271102  135563            Olesya Nikolayevna Zykina   F 19.00  171.00   64.00   
271103  135563            Olesya Nikolayevna Zykina   F 23.00  171.00   64.00   

                  Team  NOC        Games  Year  Season       City       Sport  \
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris  Tug-Of-War   
37             Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   
38             Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   
41             Finland  FIN  1948 Summer  1948  Summer     London  Gymnastics   
42             Finland  FIN  1948 Summer  1948  Summer     London  Gymnastics   
...                ...  ...          ...   ...     ...        ...         ...   
271078    Soviet Union  URS  1956 Summer  1956  Summer  Melbourne   Athletics   
271080    Soviet Union  URS  1964 Summer  1964  Summer      Tokyo   Athletics   
271082          Poland  POL  1980 Summer  1980  Summer     Moskva     Fencing   
271102          Russia  RUS  2000 Summer  2000  Summer     Sydney   Athletics   
271103          Russia  RUS  2004 Summer  2004  Summer     Athina   Athletics   

                                         Event   Medal  
3                  Tug-Of-War Men's Tug-Of-War    Gold  
37      Swimming Men's 200 metres Breaststroke  Bronze  
38      Swimming Men's 400 metres Breaststroke  Bronze  
41      Gymnastics Men's Individual All-Around  Bronze  
42            Gymnastics Men's Team All-Around    Gold  
...                                        ...     ...  
271078              Athletics Women's Shot Put  Silver  
271080              Athletics Women's Shot Put  Bronze  
271082                Fencing Men's Foil, Team  Bronze  
271102  Athletics Women's 4 x 400 metres Relay  Bronze  
271103  Athletics Women's 4 x 400 metres Relay  Silver  

[34088 rows x 15 columns]

In [49]:
df_event_mundo = df_medal_mundo.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'Games', 'Season', 'City', 'Sport', 'Name', 'ID'], axis=1)
df_event_mundo

NOC  Year                                   Event   Medal
3       DEN  1900             Tug-Of-War Men's Tug-Of-War    Gold
37      FIN  1920  Swimming Men's 200 metres Breaststroke  Bronze
38      FIN  1920  Swimming Men's 400 metres Breaststroke  Bronze
41      FIN  1948  Gymnastics Men's Individual All-Around  Bronze
42      FIN  1948        Gymnastics Men's Team All-Around    Gold
...     ...   ...                                     ...     ...
271078  URS  1956              Athletics Women's Shot Put  Silver
271080  URS  1964              Athletics Women's Shot Put  Bronze
271082  POL  1980                Fencing Men's Foil, Team  Bronze
271102  RUS  2000  Athletics Women's 4 x 400 metres Relay  Bronze
271103  RUS  2004  Athletics Women's 4 x 400 metres Relay  Silver

[34088 rows x 4 columns]

In [50]:
df_eventouro_mundo = df_event_mundo.drop_duplicates(['Event'])[df_event_mundo['Medal'] == 'Gold'].groupby(['NOC'])['Medal'].count().reset_index()
df_eventouro_mundo

<ipython-input-50-6be75f6e8ae0>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_eventouro_mundo = df_event_mundo.drop_duplicates(['Event'])[df_event_mundo['Medal'] == 'Gold'].groupby(['NOC'])['Medal'].count().reset_index()


NOC  Medal
0   AUS      2
1   AUT      3
2   AZE      1
3   BEL      8
4   BLR      3
5   BRA      1
6   BUL      2
7   CAN      5
8   CHN      6
9   CMR      1
10  DEN      4
11  ESP      7
12  ETH      2
13  EUN      1
14  FIN      3
15  FRA     26
16  GBR     23
17  GDR      2
18  GER      8
19  GRE      4
20  HUN      1
21  INA      1
22  ITA     12
23  JOR      1
24  JPN      3
25  KAZ      1
26  LTU      1
27  MAR      1
28  NED      2
29  NGR      1
30  NOR      5
31  NZL      4
32  ROU      1
33  RUS      5
34  SUI      6
35  SWE     18
36  TPE      1
37  TUR      1
38  UKR      2
39  URS      5
40  USA     41
41  UZB      1

In [51]:
print('Esse foi o país com mais medalhas de ouro da história das olimpíadas de verão:\n',df_eventouro_mundo[df_eventouro_mundo['Medal'] == df_eventouro_mundo['Medal'].max()]['NOC'].tolist())

Esse foi o país com mais medalhas de ouro da história das olimpíadas de verão:
 ['USA']


Qual país foi o maior medalhista em medalhas totais na história das olimpíadas de verão?

In [52]:
df_sport_mundo = df_medal_mundo.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'Games', 'Season', 'City', 'Event', 'Name', 'ID'], axis=1)
df_sport_mundo

NOC  Year       Sport   Medal
3       DEN  1900  Tug-Of-War    Gold
37      FIN  1920    Swimming  Bronze
38      FIN  1920    Swimming  Bronze
41      FIN  1948  Gymnastics  Bronze
42      FIN  1948  Gymnastics    Gold
...     ...   ...         ...     ...
271078  URS  1956   Athletics  Silver
271080  URS  1964   Athletics  Bronze
271082  POL  1980     Fencing  Bronze
271102  RUS  2000   Athletics  Bronze
271103  RUS  2004   Athletics  Silver

[34088 rows x 4 columns]

In [53]:
df_sportmed_mundo = df_sport_mundo.drop_duplicates(['Year', 'Sport']).groupby(['NOC'])['Medal'].count().reset_index()

df_sportmed_mundo

NOC  Medal
0   ALG      1
1   ANZ      2
2   ARG      5
3   ARM      1
4   AUS     15
..  ...    ...
74  URS     48
75  URU      3
76  USA     77
77  UZB      2
78  YUG      6

[79 rows x 2 columns]

In [54]:
print('O país que mais ganhou medalhas em quantidade total de medalhas de ouro nas olimpíadas de verão foi:\n',df_sportmed_mundo[df_sportmed_mundo['Medal'] == df_sportmed_mundo['Medal'].max()]['NOC'].tolist())

O país que mais ganhou medalhas em quantidade total de medalhas de ouro nas olimpíadas de verão foi:
 ['USA']


Obtenha o total de medalhas de ouro, prata e total por edição das Olimpíadas de verão. Lembre-se da questão do evento esportivo.

In [55]:
df_sport_mundo.groupby(['Year'])['Medal'].count()
#df_eventouro_mundo

Year
1896     143
1900     604
1904     486
1906     458
1908     831
1912     941
1920    1308
1924     832
1928     734
1932     647
1936     917
1948     852
1952     897
1956     893
1960     911
1964    1029
1968    1057
1972    1215
1976    1320
1980    1384
1984    1476
1988    1582
1992    1712
1996    1842
2000    2004
2004    2001
2008    2048
2012    1941
2016    2023
Name: Medal, dtype: int64

In [56]:
medalano_mundo = df_sport_mundo.groupby(['Year', 'Medal'])['Medal'].size().to_frame(name = 'Qtd').reset_index()

medalano_mundo

Year   Medal  Qtd
0   1896  Bronze   38
1   1896    Gold   62
2   1896  Silver   43
3   1900  Bronze  175
4   1900    Gold  201
..   ...     ...  ...
82  2012    Gold  632
83  2012  Silver  630
84  2016  Bronze  703
85  2016    Gold  665
86  2016  Silver  655

[87 rows x 3 columns]

In [57]:
medalano_mundo_pivot = medalano_mundo.pivot(index = 'Year', columns = 'Medal', values = 'Qtd')

medalano_mundo_pivot = medalano_mundo_pivot.reindex(columns = ['Gold', 'Silver', 'Bronze', 'Total'])

medalano_mundo_pivot[['Gold', 'Silver', 'Bronze']] = medalano_mundo_pivot[['Gold', 'Silver', 'Bronze']].fillna(value = 0)

medalano_mundo_pivot['Total'] = medalano_mundo_pivot['Gold'] + medalano_mundo_pivot['Silver'] + medalano_mundo_pivot['Bronze']

medalano_mundo_pivot = medalano_mundo_pivot.drop(['Bronze'], axis=1)

medalano_mundo_pivot

Medal  Gold  Silver  Total
Year                      
1896     62      43    143
1900    201     228    604
1904    173     163    486
1906    157     156    458
1908    294     281    831
1912    326     315    941
1920    493     448   1308
1924    277     281    832
1928    245     239    734
1932    229     214    647
1936    312     310    917
1948    289     284    852
1952    306     291    897
1956    302     293    893
1960    309     294    911
1964    347     339   1029
1968    359     340   1057
1972    404     392   1215
1976    438     434   1320
1980    457     458   1384
1984    497     477   1476
1988    520     513   1582
1992    559     549   1712
1996    608     605   1842
2000    663     661   2004
2004    664     660   2001
2008    671     667   2048
2012    632     630   1941
2016    665     655   2023

## 3. Brasil vs Mundo

Para finalizar, vamos fazer algumas comparações entre Brasil e mundo. Qual o ranking do Brasil em cada edição das olimpíadas? Lembrando que o ranking é ordenado por medalhas de ouro.

In [58]:
rank_mundo = df_sport_mundo.groupby(['Year', 'NOC', 'Medal'])['Medal'].size().to_frame(name = 'Qtd').reset_index()

rank_mundo

Year  NOC   Medal  Qtd
0     1896  AUS  Bronze    1
1     1896  AUS    Gold    2
2     1896  AUT  Bronze    2
3     1896  AUT    Gold    2
4     1896  AUT  Silver    1
...    ...  ...     ...  ...
2804  2016  UZB  Silver    2
2805  2016  VEN  Bronze    2
2806  2016  VEN  Silver    1
2807  2016  VIE    Gold    1
2808  2016  VIE  Silver    1

[2809 rows x 4 columns]

In [59]:
rank_mundo_pivot = rank_mundo.pivot(index = ['Year', 'NOC'], columns = 'Medal', values = 'Qtd')

rank_mundo_pivot = rank_mundo_pivot.reindex(columns = ['Gold', 'Silver', 'Bronze', 'Total'])

rank_mundo_pivot[['Gold', 'Silver', 'Bronze']] = rank_mundo_pivot[['Gold', 'Silver', 'Bronze']].fillna(value = 0)

rank_mundo_pivot['Total'] = rank_mundo_pivot['Gold'] + rank_mundo_pivot['Silver'] + rank_mundo_pivot['Bronze']

rank_mundo_pivot

Medal      Gold  Silver  Bronze  Total
Year NOC                              
1896 AUS   2.00    0.00    1.00   3.00
     AUT   2.00    1.00    2.00   5.00
     DEN   1.00    2.00    3.00   6.00
     FRA   5.00    4.00    2.00  11.00
     GBR   3.00    3.00    3.00   9.00
...         ...     ...     ...    ...
2016 UKR   2.00    8.00    5.00  15.00
     USA 139.00   54.00   71.00 264.00
     UZB   4.00    2.00    7.00  13.00
     VEN   0.00    1.00    2.00   3.00
     VIE   1.00    1.00    0.00   2.00

[1275 rows x 4 columns]

In [60]:
rank_mundo_pivot.index

MultiIndex([(1896, 'AUS'),
            (1896, 'AUT'),
            (1896, 'DEN'),
            (1896, 'FRA'),
            (1896, 'GBR'),
            (1896, 'GER'),
            (1896, 'GRE'),
            (1896, 'HUN'),
            (1896, 'SUI'),
            (1896, 'USA'),
            ...
            (2016, 'TPE'),
            (2016, 'TTO'),
            (2016, 'TUN'),
            (2016, 'TUR'),
            (2016, 'UAE'),
            (2016, 'UKR'),
            (2016, 'USA'),
            (2016, 'UZB'),
            (2016, 'VEN'),
            (2016, 'VIE')],
           names=['Year', 'NOC'], length=1275)

In [61]:
rank_mundo_ouro = rank_mundo_pivot.sort_values(by=['Year', 'Gold'], ascending = [True, False]).reset_index()
rank_mundo_ouro

Medal  Year  NOC  Gold  Silver  Bronze  Total
0      1896  GER 25.00    5.00    2.00  32.00
1      1896  USA 11.00    7.00    2.00  20.00
2      1896  GRE 10.00   18.00   20.00  48.00
3      1896  FRA  5.00    4.00    2.00  11.00
4      1896  GBR  3.00    3.00    3.00   9.00
...     ...  ...   ...     ...     ...    ...
1270   2016  QAT  0.00    1.00    0.00   1.00
1271   2016  TTO  0.00    0.00    1.00   1.00
1272   2016  TUN  0.00    0.00    3.00   3.00
1273   2016  UAE  0.00    0.00    1.00   1.00
1274   2016  VEN  0.00    1.00    2.00   3.00

[1275 rows x 6 columns]

In [62]:
#print(rank_mundo_ouro[rank_mundo_ouro['NOC']== 'BRA'].index)
#print(rank_mundo_ouro[rank_mundo_ouro['NOC']== 'BRA']['Year'].tolist())

rank_brasil_ouro = pd.DataFrame(data=rank_mundo_ouro[rank_mundo_ouro['NOC']== 'BRA'].index, index=rank_mundo_ouro[rank_mundo_ouro['NOC']== 'BRA']['Year'].tolist(), columns=['Posicao BRA'])
rank_brasil_ouro

Posicao BRA
1920          115
1948          273
1952          307
1956          347
1960          393
1964          439
1968          486
1972          525
1976          575
1980          600
1984          644
1988          693
1992          732
1996          804
2000          920
2004          957
2008         1031
2012         1117
2016         1194

Compare o maior medalhista em ouros do Brasil com o maior medalhista em ouros do mundo.

In [63]:
print('O maior medalhista do mundo em quantidade total de medalhas de ouro nas olimpíadas de verão foi:\n',df_verao[df_verao['ID'].isin(ID_ouromundo)]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com', df_verao[df_verao['Medal'] == 'Gold'].groupby(['ID']).size()[(df_verao[df_verao['Medal'] == 'Gold'].groupby(['ID']).size()).values == df_verao[df_verao['Medal'] == 'Gold'].groupby(['ID']).size().max()].values, 'medalhas.')

print('Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:\n',df_bm[df_bm['ID'].isin(ID_ouro)]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com', df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size().max(), 'medalhas cada.')

O maior medalhista do mundo em quantidade total de medalhas de ouro nas olimpíadas de verão foi:
 ['Michael Fred Phelps, II'] com [23] medalhas.
Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:
 ['Adhemar Ferreira da Silva', 'Srgio "Escadinha" Dutra dos Santos', 'Fabiana "Fabi" Alvim de Oliveira', 'Fabiana Marcelino Claudino', 'Marcelo Bastos Ferreira', 'Giovane Farinazzo Gvio', 'Torben Schmidt Grael', 'Jaqueline Maria "Jaque" Pereira de Carvalho Endres', 'Maurcio Camargo Lima', 'Paula Renata Marques Pequeno', 'Robert Scheidt', 'Sheilla Tavares de Castro Blassioli', 'Thasa Daher de Menezes'] com 2 medalhas cada.


Compare o maior medalhista em total de medalhas do Brasil com o maior medalhista em total de medalhas do mundo.

In [64]:
print('O maiores medalhista do mundo em quantidade total de medalhas das olimpíadas de verão foi:\n',df_verao[df_verao['ID'].isin(ID_medmundo['ID'])]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com', ID_medmundo[0].values, 'medalhas.')

print('Os maiores medalhistas brasileiros em quantidade total de medalhas foram:\n',df_bm[df_bm['ID'].isin(ID_maismed['ID'])]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com',ID_maismed[0].mean(),'medalhas cada.')

O maiores medalhista do mundo em quantidade total de medalhas das olimpíadas de verão foi:
 ['Robert Tait McKenzie'] com [58] medalhas.
Os maiores medalhistas brasileiros em quantidade total de medalhas foram:
 ['Torben Schmidt Grael', 'Robert Scheidt'] com 5.0 medalhas cada.


Compare o maior medalhista em ouros do Brasil com o maior medalhista do mundo no mesmo esporte.

In [65]:
print('Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:\n',df_bm[df_bm['ID'].isin(ID_ouro)]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com', df_bm[df_bm['Medal'] == 'Gold'].groupby(['ID']).size().max(), 'medalhas cada.\n')

df_bm_sport = df_brazil_medal.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games', 'Year', 'Season', 'City', 'Event'], axis=1)

print('Esses são os esportes nas quais temos os maiores medalhistas brasileiros em quantidade total de medalhas de ouro:', df_bm_sport[df_bm_sport['ID'].isin(ID_ouro)]['Sport'].drop_duplicates().reset_index()['Sport'].tolist())

Os maiores medalhistas brasileiros em quantidade total de medalhas de ouro foram:
 ['Adhemar Ferreira da Silva', 'Srgio "Escadinha" Dutra dos Santos', 'Fabiana "Fabi" Alvim de Oliveira', 'Fabiana Marcelino Claudino', 'Marcelo Bastos Ferreira', 'Giovane Farinazzo Gvio', 'Torben Schmidt Grael', 'Jaqueline Maria "Jaque" Pereira de Carvalho Endres', 'Maurcio Camargo Lima', 'Paula Renata Marques Pequeno', 'Robert Scheidt', 'Sheilla Tavares de Castro Blassioli', 'Thasa Daher de Menezes'] com 2 medalhas cada.

Esses são os esportes nas quais temos os maiores medalhistas brasileiros em quantidade total de medalhas de ouro: ['Athletics', 'Volleyball', 'Sailing']


In [66]:
mundo_ouro = df_medal_mundo.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'Games', 'Season', 'City', 'Event', 'NOC'], axis=1)
df_sportouro_mundo = mundo_ouro[mundo_ouro['Medal'] == 'Gold'].groupby(['Sport','ID'])['Medal'].count().to_frame(name = 'Qtd').reset_index()

#df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Athletics']['Qtd'].max()
#df_sportouro_mundo[(df_sportouro_mundo['Sport'].isin(['Athletics', 'Volleyball', 'Sailing'])) & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'].isin(['Athletics', 'Volleyball', 'Sailing'])]['Qtd'].max())]['ID'].tolist()
#df_sportouro_mundo.groupby(['Sport', 'ID']).size()

id_athletics = df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Athletics') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Athletics']['Qtd'].max())]['ID'].tolist()
id_volleyball = df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Athletics') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Volleyball']['Qtd'].max())]['ID'].tolist()
id_sailing = df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Athletics') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Sailing']['Qtd'].max())]['ID'].tolist()


print("Esse foi o maior medalhista do esporte 'Athletics':", mundo_ouro[mundo_ouro['ID'].isin(id_athletics)]['Name'].drop_duplicates().values,'com',df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Athletics') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Athletics']['Qtd'].max())]['Qtd'].drop_duplicates().tolist(),'medalhas.\n')
print("Esses foram os maiores medalhistas do esporte 'Volleyball':\n", mundo_ouro[mundo_ouro['ID'].isin(id_volleyball)]['Name'].drop_duplicates().values,'\n','com',df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Volleyball') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Volleyball']['Qtd'].max())]['Qtd'].drop_duplicates().tolist(),'medalhas cada.\n')
print("Esses foram os maiores medalhistas do esporte 'Sailing':\n", mundo_ouro[mundo_ouro['ID'].isin(id_sailing)]['Name'].drop_duplicates().values,'com',df_sportouro_mundo[(df_sportouro_mundo['Sport']== 'Sailing') & (df_sportouro_mundo['Qtd'] == df_sportouro_mundo[df_sportouro_mundo['Sport'] == 'Sailing']['Qtd'].max())]['Qtd'].drop_duplicates().tolist(),'medalhas cada.\n')

Esse foi o maior medalhista do esporte 'Athletics': ['Raymond Clarence "Ray" Ewry'] com [10] medalhas.

Esses foram os maiores medalhistas do esporte 'Volleyball':
 ['Kenenisa Bekele Beyecha' 'Valerie Ann Brisco-Hooks'
 'Olha Arkadivna Bryzhina (Vladykina-)' 'Veronica Angella Campbell-Brown'
 'Glenn Ashby Davis' 'Yolanda Gail L. Devers (-Roberts)'
 'Tirunesh Dibaba Keneni' 'John Joseph Flanagan' 'Ugo Frigerio'
 'Delorez Florence Griffith Joyner' 'Harry Livingston Hillman, Jr.'
 'Lemuel Clarence "Bud" Houser' 'Jacqueline "Jackie" Joyner-Kersee'
 'Tatyana Vasilyevna Kazankina (-Kovalenko)' 'Steven Earl "Steve" Lewis'
 'Tianna Madison-Bartoletta' 'LaShawn Merritt' 'Bobby Joe Morrow'
 'Marie-Jos Juliana Prec (Hontas-)' 'Paul Henry Pilgrim'
 'Tamara Natanovna Press' 'Ralph Waldo Rose'
 'Wilma Glodean Rudolph (-Ward, -Eldridge)' 'Viktor Saneyev'
 'Peter George Snell' 'Renate Stecher (Meiner-)'
 'Shirley Barbara Strickland de la Hunty' 'Irena Szewiska-Kirszenstein'
 'Angelo F. Taylor, Jr.'
 '

Compare o maior medalhista em total de medalhas do Brasil com o maior medalhista do mundo no mesmo esporte.

In [67]:
df_bm_sport

ID                                          Name  \
1651       918                           Ademir Roque Kaefer   
1652       918                           Ademir Roque Kaefer   
1668       925           Adenzia Aparecida Ferreira da Silva   
1733       966                                  Daniel Adler   
1856      1020                  Adriana Aparecida dos Santos   
...        ...                                           ...   
267709  133931                      Arthur Nabarrete Zanetti   
268211  134186              Jos Elias "Z Elias" Moedim Jnior   
268218  134190  Jos Marco "Z Marco" Nobrega Ferreira de Melo   
268219  134191                Jos Marcelo "Z Maria" Ferreira   
268247  134211                        Jos Carlos Cracco Neto   

                   Sport   Medal  
1651            Football  Silver  
1652            Football  Silver  
1668          Volleyball    Gold  
1733             Sailing  Silver  
1856          Basketball  Silver  
...                  ...     ...  
267709        Gymnastics  Silver  
268211          Football  Bronze  
268218  Beach Volleyball  Silver  
268219          Football  Bronze  
268247          Football    Gold  

[475 rows x 4 columns]

In [68]:
print('Os maiores medalhistas brasileiros em quantidade total de medalhas foram:\n',df_bm[df_bm['ID'].isin(ID_maismed['ID'])]['Name'].drop_duplicates().reset_index()['Name'].tolist(),'com', ID_maismed[0].drop_duplicates().values, 'medalhas cada.\n')

print('Esse é o esportes em qual temos os maiores medalhistas brasileiros em quantidade total de medalhas:', df_bm_sport[df_bm_sport['ID'].isin(ID_maismed['ID'])]['Sport'].drop_duplicates().reset_index()['Sport'].tolist())


Os maiores medalhistas brasileiros em quantidade total de medalhas foram:
 ['Torben Schmidt Grael', 'Robert Scheidt'] com [5] medalhas cada.

Esse é o esportes em qual temos os maiores medalhistas brasileiros em quantidade total de medalhas: ['Sailing']


In [69]:
mundo_maismed = df_medal_mundo.drop(['Sex', 'Age', 'Height', 'Weight', 'Team', 'Games', 'Season', 'City', 'Event', 'NOC'], axis=1)
df_sportmaismed_mundo = mundo_maismed.groupby(['Sport','ID'])['Medal'].count().to_frame(name = 'Qtd').reset_index()



id_maismed_sailing = df_sportmaismed_mundo[(df_sportmaismed_mundo['Sport'] == 'Sailing') & (df_sportmaismed_mundo['Qtd']== df_sportmaismed_mundo[df_sportmaismed_mundo['Sport'] == 'Sailing']['Qtd'].max())]['ID'].tolist()

print("Esses foram os maiores medalhistas do esporte 'Sailing':\n", mundo_maismed[mundo_maismed['ID'].isin(id_maismed_sailing)]['Name'].drop_duplicates().values,'com',df_sportmaismed_mundo[(df_sportmaismed_mundo['Sport']== 'Sailing') & (df_sportmaismed_mundo['Qtd'] == df_sportmaismed_mundo[df_sportmaismed_mundo['Sport'] == 'Sailing']['Qtd'].max())]['Qtd'].drop_duplicates().tolist(),'medalhas cada.\n')

Esses foram os maiores medalhistas do esporte 'Sailing':
 ['Charles Benedict "Ben" Ainslie' 'Torben Schmidt Grael' 'Robert Scheidt'] com [5] medalhas cada.



Calcule o percentual de medalhas de ouro, prata e bronze que o Brasil ganhou em cada olimpíada.

In [79]:
result_bra = rank_mundo_ouro[rank_mundo_ouro['NOC']=='BRA']
result_mundo = rank_mundo_pivot.reset_index()

In [80]:
result_bra

Medal  Year  NOC  Gold  Silver  Bronze  Total
115    1920  BRA  1.00    1.00    5.00   7.00
273    1948  BRA  0.00    0.00   10.00  10.00
307    1952  BRA  1.00    0.00    2.00   3.00
347    1956  BRA  1.00    0.00    0.00   1.00
393    1960  BRA  0.00    0.00   13.00  13.00
439    1964  BRA  0.00    0.00   12.00  12.00
486    1968  BRA  0.00    1.00    3.00   4.00
525    1972  BRA  0.00    0.00    2.00   2.00
575    1976  BRA  0.00    0.00    3.00   3.00
600    1980  BRA  4.00    0.00    5.00   9.00
644    1984  BRA  1.00   33.00    2.00  36.00
693    1988  BRA  1.00   17.00    5.00  23.00
732    1992  BRA 13.00    1.00    0.00  14.00
804    1996  BRA  5.00   15.00   43.00  63.00
920    2000  BRA  0.00   12.00   36.00  48.00
957    2004  BRA 18.00   19.00    3.00  40.00
1031   2008  BRA 14.00   34.00   30.00  78.00
1117   2012  BRA 14.00   34.00   11.00  59.00
1194   2016  BRA 36.00    8.00    6.00  50.00

In [81]:
result_mundo

Medal  Year  NOC   Gold  Silver  Bronze  Total
0      1896  AUS   2.00    0.00    1.00   3.00
1      1896  AUT   2.00    1.00    2.00   5.00
2      1896  DEN   1.00    2.00    3.00   6.00
3      1896  FRA   5.00    4.00    2.00  11.00
4      1896  GBR   3.00    3.00    3.00   9.00
...     ...  ...    ...     ...     ...    ...
1270   2016  UKR   2.00    8.00    5.00  15.00
1271   2016  USA 139.00   54.00   71.00 264.00
1272   2016  UZB   4.00    2.00    7.00  13.00
1273   2016  VEN   0.00    1.00    2.00   3.00
1274   2016  VIE   1.00    1.00    0.00   2.00

[1275 rows x 6 columns]

In [90]:
result_mundo[~result_mundo['Year'].isin(result_bra['Year'])]['Year'].drop_duplicates().reset_index()

index  Year
0      0  1896
1     10  1900
2     34  1904
3     46  1906
4     64  1908
5     83  1912
6    124  1924
7    155  1928
8    189  1932
9    217  1936

In [108]:
result_bra2 = pd.concat([result_bra.reset_index(), result_mundo[~result_mundo['Year'].isin(result_bra['Year'])]['Year'].drop_duplicates().reset_index()], keys='Year')
result_bra2 = result_bra2.drop(columns = ['index'], axis=1).fillna(value = 0)
result_bra2['NOC'] = 'BRA'
result_bra2 = result_bra2.sort_values(['Year'], ascending=True)
result_bra2 = result_bra2.reset_index().drop(columns=['level_0', 'level_1'],axis=1)
result_bra2

Year  NOC  Gold  Silver  Bronze  Total
0   1896  BRA  0.00    0.00    0.00   0.00
1   1900  BRA  0.00    0.00    0.00   0.00
2   1904  BRA  0.00    0.00    0.00   0.00
3   1906  BRA  0.00    0.00    0.00   0.00
4   1908  BRA  0.00    0.00    0.00   0.00
5   1912  BRA  0.00    0.00    0.00   0.00
6   1920  BRA  1.00    1.00    5.00   7.00
7   1924  BRA  0.00    0.00    0.00   0.00
8   1928  BRA  0.00    0.00    0.00   0.00
9   1932  BRA  0.00    0.00    0.00   0.00
10  1936  BRA  0.00    0.00    0.00   0.00
11  1948  BRA  0.00    0.00   10.00  10.00
12  1952  BRA  1.00    0.00    2.00   3.00
13  1956  BRA  1.00    0.00    0.00   1.00
14  1960  BRA  0.00    0.00   13.00  13.00
15  1964  BRA  0.00    0.00   12.00  12.00
16  1968  BRA  0.00    1.00    3.00   4.00
17  1972  BRA  0.00    0.00    2.00   2.00
18  1976  BRA  0.00    0.00    3.00   3.00
19  1980  BRA  4.00    0.00    5.00   9.00
20  1984  BRA  1.00   33.00    2.00  36.00
21  1988  BRA  1.00   17.00    5.00  23.00
22  1992  BRA 13.00    1.00    0.00  14.00
23  1996  BRA  5.00   15.00   43.00  63.00
24  2000  BRA  0.00   12.00   36.00  48.00
25  2004  BRA 18.00   19.00    3.00  40.00
26  2008  BRA 14.00   34.00   30.00  78.00
27  2012  BRA 14.00   34.00   11.00  59.00
28  2016  BRA 36.00    8.00    6.00  50.00

In [119]:
result_mundo2 = result_mundo.groupby('Year').sum().reset_index()
result_mundo2

Medal  Year   Gold  Silver  Bronze    Total
0      1896  62.00   43.00   38.00   143.00
1      1900 201.00  228.00  175.00   604.00
2      1904 173.00  163.00  150.00   486.00
3      1906 157.00  156.00  145.00   458.00
4      1908 294.00  281.00  256.00   831.00
5      1912 326.00  315.00  300.00   941.00
6      1920 493.00  448.00  367.00 1,308.00
7      1924 277.00  281.00  274.00   832.00
8      1928 245.00  239.00  250.00   734.00
9      1932 229.00  214.00  204.00   647.00
10     1936 312.00  310.00  295.00   917.00
11     1948 289.00  284.00  279.00   852.00
12     1952 306.00  291.00  300.00   897.00
13     1956 302.00  293.00  298.00   893.00
14     1960 309.00  294.00  308.00   911.00
15     1964 347.00  339.00  343.00 1,029.00
16     1968 359.00  340.00  358.00 1,057.00
17     1972 404.00  392.00  419.00 1,215.00
18     1976 438.00  434.00  448.00 1,320.00
19     1980 457.00  458.00  469.00 1,384.00
20     1984 497.00  477.00  502.00 1,476.00
21     1988 520.00  513.00  549.00 1,582.00
22     1992 559.00  549.00  604.00 1,712.00
23     1996 608.00  605.00  629.00 1,842.00
24     2000 663.00  661.00  680.00 2,004.00
25     2004 664.00  660.00  677.00 2,001.00
26     2008 671.00  667.00  710.00 2,048.00
27     2012 632.00  630.00  679.00 1,941.00
28     2016 665.00  655.00  703.00 2,023.00

In [163]:
result_bra2['Gold%'] = ((result_bra2['Gold'])/(result_mundo2['Gold'])*100).map("{:,.2f}%".format)
result_bra2['Silver%'] = ((result_bra2['Silver'])/(result_mundo2['Silver'])*100).map("{:,.2f}%".format)
result_bra2['Bronze%'] = ((result_bra2['Bronze'])/(result_mundo2['Bronze'])*100).map("{:,.2f}%".format)
result_bra2['Total%'] = ((result_bra2['Total'])/(result_mundo2['Total'])*100).map("{:,.2f}%".format)

result_bra2

Year  NOC  Gold  Silver  Bronze  Total  Gold% Silver% Bronze% Total%
0   1896  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
1   1900  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
2   1904  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
3   1906  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
4   1908  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
5   1912  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
6   1920  BRA  1.00    1.00    5.00   7.00  0.20%   0.22%   1.36%  0.54%
7   1924  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
8   1928  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
9   1932  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
10  1936  BRA  0.00    0.00    0.00   0.00  0.00%   0.00%   0.00%  0.00%
11  1948  BRA  0.00    0.00   10.00  10.00  0.00%   0.00%   3.58%  1.17%
12  1952  BRA  1.00    0.00    2.00   3.00  0.33%   0.00%   0.67%  0.33%
13  1956  BRA  1.00    0.00    0.00   1.00  0.33%   0.00%   0.00%  0.11%
14  1960  BRA  0.00    0.00   13.00  13.00  0.00%   0.00%   4.22%  1.43%
15  1964  BRA  0.00    0.00   12.00  12.00  0.00%   0.00%   3.50%  1.17%
16  1968  BRA  0.00    1.00    3.00   4.00  0.00%   0.29%   0.84%  0.38%
17  1972  BRA  0.00    0.00    2.00   2.00  0.00%   0.00%   0.48%  0.16%
18  1976  BRA  0.00    0.00    3.00   3.00  0.00%   0.00%   0.67%  0.23%
19  1980  BRA  4.00    0.00    5.00   9.00  0.88%   0.00%   1.07%  0.65%
20  1984  BRA  1.00   33.00    2.00  36.00  0.20%   6.92%   0.40%  2.44%
21  1988  BRA  1.00   17.00    5.00  23.00  0.19%   3.31%   0.91%  1.45%
22  1992  BRA 13.00    1.00    0.00  14.00  2.33%   0.18%   0.00%  0.82%
23  1996  BRA  5.00   15.00   43.00  63.00  0.82%   2.48%   6.84%  3.42%
24  2000  BRA  0.00   12.00   36.00  48.00  0.00%   1.82%   5.29%  2.40%
25  2004  BRA 18.00   19.00    3.00  40.00  2.71%   2.88%   0.44%  2.00%
26  2008  BRA 14.00   34.00   30.00  78.00  2.09%   5.10%   4.23%  3.81%
27  2012  BRA 14.00   34.00   11.00  59.00  2.22%   5.40%   1.62%  3.04%
28  2016  BRA 36.00    8.00    6.00  50.00  5.41%   1.22%   0.85%  2.47%